pkg/apis/org/v1/che_types.go

```diff
@@ -81,6 +81,8 @@ type CheClusterSpecServer struct {
 	// By default a dedicated plugin registry server is started.
 	// But if ExternalPluginRegistry is `true`, then no such dedicated server will be started by the operator
 	ExternalPluginRegistry bool `json:"externalPluginRegistry"`
+	// OverrideCheProperties is a list of EnvVars of Che properties to override from their default value
+	OverrideCheProperties []ChePropertyOverride `json:"overrideCheProperties"`
```

Comment:
Did you try using a `map[string]string` ?

That could:
- avoid the risk of adding several times the same properties with different values
- simplify some parts of your code.